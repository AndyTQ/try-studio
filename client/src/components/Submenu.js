import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider';
import ListSubheader from '@material-ui/core/ListSubheader';
import SpotifyIcon from '@material-ui/icons/Contactless';
import AppleIcon from '@material-ui/icons/Apple';
import YouTubeIcon from '@material-ui/icons/YouTube';
import RecentIcon from '@material-ui/icons/QueryBuilder';
import FavoriteIcon from '@material-ui/icons/Favorite';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
  },
}));

const items = [
  { icon: <SpotifyIcon />, name: "Spotify", enabled: true },
  { icon: <AppleIcon />, name: "Apple Music (Coming)", enabled: false },
  { icon: <YouTubeIcon />,  name: "PlaYoutube Music (Coming)", enabled: false },
];

const streamingServicePanel = () => {
  return(
    <div>
        {items.map(item => (
          <ListItem button disabled={!item.enabled}>
            <ListItemIcon>
              {item.icon}
            </ListItemIcon>
            <ListItemText primary={item.name} />
          </ListItem>
        ))}
    </div>
  );
}


export default function Submenu() {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <List
        subheader={
          <ListSubheader component="div">
            Streaming Service
          </ListSubheader>
        }
        component="nav"
        aria-label="main mailbox folders">
        {streamingServicePanel()}
      </List>
      <Divider />
    </div>
  );
}