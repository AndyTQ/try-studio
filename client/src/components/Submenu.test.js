import React from "react";
import { create } from "react-test-renderer";
import Submenu from "./Submenu";

describe("Submenu component", () => {
  test("Matches the snapshot", () => {
    const submenu = create(<Submenu />);
    expect(submenu.toJSON()).toMatchSnapshot();
  });
});