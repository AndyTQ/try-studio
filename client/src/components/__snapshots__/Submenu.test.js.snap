// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submenu component Matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <nav
    aria-label="main mailbox folders"
    className="MuiList-root MuiList-padding MuiList-subheader"
  >
    <div
      className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      Streaming Service
    </div>
    <div>
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8.46 14.45l-1.36-.62c.28-.61.41-1.24.4-1.86-.01-.63-.14-1.24-.4-1.8l1.36-.63c.35.75.53 1.56.54 2.4.01.86-.17 1.7-.54 2.51zm3.07 1.56l-1.3-.74c.52-.92.78-1.98.78-3.15 0-1.19-.27-2.33-.8-3.4l1.34-.67c.64 1.28.96 2.65.96 4.07 0 1.43-.33 2.74-.98 3.89zm3.14 1.32l-1.35-.66c.78-1.6 1.18-3.18 1.18-4.69 0-1.51-.4-3.07-1.18-4.64l1.34-.67c.9 1.78 1.34 3.56 1.34 5.31 0 1.74-.44 3.54-1.33 5.35z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Spotify
          </span>
        </div>
      </div>
      <div
        aria-disabled={true}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters Mui-disabled MuiListItem-button Mui-disabled"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.71 19.5c-.83 1.24-1.71 2.45-3.05 2.47-1.34.03-1.77-.79-3.29-.79-1.53 0-2 .77-3.27.82-1.31.05-2.3-1.32-3.14-2.53C4.25 17 2.94 12.45 4.7 9.39c.87-1.52 2.43-2.48 4.12-2.51 1.28-.02 2.5.87 3.29.87.78 0 2.26-1.07 3.81-.91.65.03 2.47.26 3.64 1.98-.09.06-2.17 1.28-2.15 3.81.03 3.02 2.65 4.03 2.68 4.04-.03.07-.42 1.44-1.38 2.83M13 3.5c.73-.83 1.94-1.46 2.94-1.5.13 1.17-.34 2.35-1.04 3.19-.69.85-1.83 1.51-2.95 1.42-.15-1.15.41-2.35 1.05-3.11z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Apple Music (Coming)
          </span>
        </div>
      </div>
      <div
        aria-disabled={true}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters Mui-disabled MuiListItem-button Mui-disabled"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            PlaYoutube Music (Coming)
          </span>
        </div>
      </div>
    </div>
  </nav>
  <hr
    className="MuiDivider-root"
  />
</div>
`;
